Unix/wsman/wsbuf.c

```diff
@@ -3214,6 +3214,15 @@ static MI_Result WSBuf_CreateRequestHeader(WSBuf *buf,
             goto failed;
         }
     }
+    if (cliHeaders->sessionId)
+    {
+        if (MI_RESULT_OK != WSBuf_AddLit(buf, LIT(ZT("<p:SessionId s:mustUnderstand=\"false\">"))) ||
+            MI_RESULT_OK != WSBuf_AddStringNoEncoding(buf, cliHeaders->sessionId) ||
+            MI_RESULT_OK != WSBuf_AddLit(buf, LIT(ZT("</p:SessionId>"))))
+        {
+            goto failed;
```

Comment:
There are three different reasons that `goto failed` will get control. Do we ever care which of these failures cause the problem?

I see it's a common calling construct. I'm just wondering if, long term, that's a good approach. It depends on when/how often failures can occur ...